### ISSUE TITLE
Theme is broken on Python2

### ISSUE THREAD

**Issue by zachriggle** [pwndbg/pwndbg#117]
```
$ gdb
Loaded 94 commands.  Type pwndbg [filter] for a list.
pwndbg> theme
Traceback (most recent call last):
  File "/home/riggle/pwndbg/pwndbg/commands/__init__.py", line 54, in __call__
    return self.function(*args, **kwargs)
  File "/home/riggle/pwndbg/pwndbg/commands/theme.py", line 22, in theme
    longest_value = max(map(len, [extend_value_with_default(repr(v.value), repr(v.default)) for v in values]))
UnicodeEncodeError: 'ascii' codec can't encode characters
```

**Comment by zachriggle** [comment]
This also breaks the `context` command

**Comment by anthraxx** [comment]
Are you sure this is a python2 issue? this sounds more like a $LC_CTYPE encoding issues, doesn't it? Could dig into this tomorrow :sweat_smile:

**Comment by zachriggle** [comment]
Ultimately the issue boils down to printing Unicode characters.

```py
>>> print u'\u2713'
✓
>>> print u'\u2713'.decode()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2713' in position 0: ordinal not in range(128)
```

It's odd that this issue is appearing now, since we've always printed unicode check-marks.

**Comment by disconnect3d** [comment]
Probably the advices that can be find [here](http://python-future.org/compatible_idioms.html#strings-and-bytes) would fix the problem. I mean:
* converting string literals `"asd"` to unicode ones `u"asd"`
* `from builtins import str` as suggested [here](http://python-future.org/compatible_idioms.html#unicode), so that `str(...)` returns something that is treated as `unicode`:
```python
In [1]: type(str()), isinstance(str(), unicode)
Out[1]: (str, False)

In [2]: from builtins import str

In [3]: type(str()), isinstance(str(), unicode)
Out[3]: (future.types.newstr.newstr, True)
```

I will try to prepare patch today as I need to launch pwndbg on a server with gdb that uses Python 2.

**Comment by disconnect3d** [comment]
I have tried several things to fix the issue but I failed.

However as I still want to launch it on a gdb with Python 2, I just changed all unicode characters to more or less their ASCII counterparts.
The unicode characters can be found with `git grep`:
```bash
~/tools/pwndbg/pwndbg$ git grep --color='auto' -P -n "[\x80-\xFF]"
chain.py:43:config_arrow_left  = theme.Parameter('chain-arrow-left', '<E2><97><82><E2><80><94>', 'left arrow of chain formatting')
chain.py:44:config_arrow_right = theme.Parameter('chain-arrow-right', '<E2><80><94><E2><96><B8>', 'right arrow of chain formatting')
color/backtrace.py:12:config_prefix         = theme.Parameter('backtrace-prefix', '<E2><96><BA>', 'prefix for current backtrace label')
color/disasm.py:83:        asm = green(u'<E2><9C><94> ') + asm
commands/nearpc.py:31:nearpc_branch_marker = pwndbg.color.theme.Parameter('nearpc-branch-marker', u'    <E2><86><93>', 'branch marker line for nearpc command')
commands/nearpc.py:34:pwndbg.color.theme.Parameter('nearpc-prefix', u'<E2><96><BA>', 'prefix marker for nearpc command')
commands/telescope.py:29:offset_separator = theme.Parameter('telescope-offset-separator', u'<E2><94><82>', 'offset separator of the telescope command')
commands/telescope.py:31:repeating_maker  = theme.Parameter('telescope-repeating-marker', u'... <E2><86><93>', 'repeating values marker of the telescope command')
config.py:45:# such as <E2><80><98>\t<E2><80><99>, <E2><80><98>\f<E2><80><99>, and octal escapes, are translated into
hexdump.py:29:config_separator      = theme.Parameter('hexdump-ascii-block-separator', '<E2><94><82>', 'block separator char of the hexdump command')
ui.py:21:theme.Parameter('banner-separator', '<E2><94><80>', 'repeated banner separator character')
```

If someone needs that, below is a git patch that changes unicode characters into ASCII ones has been made from master branch (b1892b27741a478cd5361061b0b4dda9ef46d02e).
[removed_unicode_characters.patch.txt](https://github.com/pwndbg/pwndbg/files/534273/removed_unicode_characters.patch.txt)

I believe this may also help in further debugging of this issue.

**Comment by zachriggle** [comment]
We are already doing "from __future__ import unicode_literals" everywhere,
which removes the need for u prefixes.

Also, six.string_types or its equivalent is preferred to any other
compatibility modules.

**Comment by disconnect3d** [comment]
@zachriggle Yeah I have noticed it when I was debugging the issue. The above was just my first thought...
If you haven't noticed, I have posted a workaround for the issue above.

I have also tried to changing the pasted unicode characters into encoded strings of `'\u25c2\u2014'` or `'\xe2\x97\x82\xe2\x80\x94'` or even doing `.encode('utf8')` but it didn't help.

```python
In [3]: from __future__ import unicode_literals

In [4]: '◂—'
Out[4]: u'\u25c2\u2014'

In [5]: a=_

In [6]: a.encode('utf8')
Out[6]: '\xe2\x97\x82\xe2\x80\x94'
```

**Comment by anthraxx** [comment]
@disconnect3d please try my patch proposal, it should work (at least it does for my test compiled gdb with python2).

**Comment by disconnect3d** [comment]
@anthraxx Works for me.
The issue can be closed.

**Comment by zachriggle** [comment]
Closed via #129

**Comment by PetyrBaelish** [comment]
Sorry, this issue persists:
GNU gdb (Ubuntu 7.11.1-0ubuntu1~16.04) 7.11.1
...
Traceback (most recent call last):
  File "/home/u/tools/pwndbg/gdbinit.py", line 15, in <module>
    import pwndbg # isort:skip
  File "/home/u/tools/pwndbg/pwndbg/__init__.py", line 18, in <module>
    import pwndbg.commands
  File "/home/u/tools/pwndbg/pwndbg/commands/__init__.py", line 15, in <module>
    import pwndbg.chain
  File "/home/u/tools/pwndbg/pwndbg/chain.py", line 43, in <module>
    config_arrow_left  = theme.Parameter('chain-arrow-left', '◂—', 'left arrow of chain formatting')
  File "/home/u/tools/pwndbg/pwndbg/color/theme.py", line 17, in __init__
    'theme')
  File "/home/u/tools/pwndbg/pwndbg/config.py", line 83, in __init__
    self.value = default
UnicodeEncodeError: 'ascii' codec can't encode characters in position 0-1: ordinal not in range(128)
...
>(gdb) python print(sys.version)
3.5.2 (default, Sep 10 2016, 08:21:44) 
[GCC 5.4.0 20160609]
